ui/src/VoxettaUserForm.js

```diff
@@ -0,0 +1,255 @@
+            <section class="container">
+                <div class="header-container">
+                    <div class="icon-container">
+                        <mwc-icon>account_circle</mwc-icon>
+                    </div>
+                    <div class="title-container">
+                        <h1 class="title">User information</h1>
+                    </div>
+                </div>
+                <h2 class="description">
+                    Your recording will be tagged with the following:</h2>
+                <article id="user-form" class="container">
+                    <mwc-textfield 
+                        id="user-id"
+                        outlined 
+                        label="User identifier"
+                        value=${this.userId}>
+                    </mwc-textfield>
+                    <div class="mwc-select">
+                        <mwc-select
+                            id="gender-list" 
+                            outlined 
+                            label="Gender" 
+                            placeholder="Select your gender"
+                            value=${this.gender}>
+                                <mwc-list-item disabled></mwc-list-item>
+                                <mwc-list-item value="Female">Female</mwc-list-item>
+                                <mwc-list-item value="Male">Male</mwc-list-item>  
+                                <mwc-list-item value="Other">Other</mwc-list-item>     
+                        </mwc-select>
+                    <div>
+                    <mwc-textfield 
+                        id="user-age"
+                        type="number"
+                        outlined 
+                        label="Age" 
+                        placeholder="Enter your age"
+                        min=0
+                        max=120
+                        value=${this.userAge}>
+                    </mwc-textfield>
+                    <mwc-textfield 
+                        id="device-type"
+                        outlined 
+                        label="Device" 
+                        placeholder="Device type"
+                        value=${this.deviceType}>
+                    </mwc-textfield>
+                    <mwc-button 
+                        class="save"
+                        unelevated 
+                        label="Save"
+                        @click=${this.processForm}>
+                    </mwc-button>
+                </article>
+                <mwc-button 
+                    class="cancel"
+                    unelevated 
+                    label="Cancel"
+                    @click=${this.handleExitForm}>
+                </mwc-button>
+            </section>
+        `;
+    }
+
+    /**
+     * Checks to see if the user submitted valid information, and if so,
+     * saves the user information in a cookie. 
+     */
+    processForm() {
+        this.userId = this.shadowRoot.getElementById('user-id').value;
```

Comment:
mwc-textfield has built in validation https://github.com/material-components/material-components-web-components/tree/master/packages/textfield#validation

So you can use the default validators or attach custom validation logic to each input field, and also use the `validationMessage` to attach the error directly to that input. Each input will then report its `validity.valid` which you can roll up into one boolean with JavaScript

https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation

Reply:
fixed